mesonbuild/modules/python.py

```diff
@@ -255,8 +255,9 @@ def _dependency_method_impl(self, kwargs: TYPE_kwargs) -> Dependency:
     @disablerIfNotFound
     @permittedKwargs(permitted_dependency_kwargs | {'embed'})
     @FeatureNewKwargs('python_installation.dependency', '0.53.0', ['embed'])
-    @noPosargs
     @InterpreterObject.method('dependency')
+    @noPosargs
+    @typed_kwargs('python_installation.dependency', *DEPENDENCY_KWS, allow_unknown=True)
```

Comment:
I'd squash into the previous commit, personally.

Reply:
I think this is just a rebase artifact, AFAICT there's no different between putting these before or after the `InterpreterObject.method` call, so I'll just move them both above to reduce the noise

Reply:
You're right that there's a rebase artifact but that's unrelated to the previous commit, technically.